frontend/hub/execution-environments/ExecutionEnvironmentForm.tsx

```diff
@@ -0,0 +1,333 @@
+import { useTranslation, TFunction } from 'react-i18next';
+import { useParams } from 'react-router-dom';
+import {
+  PageFormSubmitHandler,
+  PageFormTextInput,
+  PageHeader,
+  PageLayout,
+  useGetPageUrl,
+  PageFormTextArea,
+} from '../../../framework';
+import { TagIcon } from '@patternfly/react-icons';
+import { Button, InputGroup, Label, LabelGroup, TextInput } from '@patternfly/react-core';
+import { PageFormGroup } from '../../../framework/PageForm/Inputs/PageFormGroup';
+import { useGet } from '../../common/crud/useGet';
+import { HubRoute } from '../HubRoutes';
+import { hubAPI, hubAPIPost, pulpAPI } from '../api/utils';
+import { ExecutionEnvironment } from './ExecutionEnvironment';
+import { HubPageForm } from '../HubPageForm';
+import { HubItemsResponse } from '../useHubView';
+import { useState, useCallback, useMemo } from 'react';
+
+import { patchHubRequest, putHubRequest } from './../api/request';
+import { PageFormAsyncSelect } from '../../../framework/PageForm/Inputs/PageFormAsyncSelect';
+import { useSelectRegistrySingle } from './hooks/useRegistrySelector';
+import { usePageNavigate } from '../../../framework';
+
+export function CreateExecutionEnvironment() {
+  return <ExecutionEnvironmentForm mode="add" />;
+}
+
+export function EditExecutionEnvironment() {
+  return <ExecutionEnvironmentForm mode="edit" />;
+}
+
+function ExecutionEnvironmentForm(props: { mode: 'add' | 'edit' }) {
+  const page_size = 50;
+  const { t } = useTranslation();
+  const navigate = usePageNavigate();
+  const getPageUrl = useGetPageUrl();
+  const mode = props.mode;
+  const params = useParams<{ id?: string }>();
+
+  const [tagsToInclude, setTagsToInclude] = useState<string[]>([]);
+  const [tagsToExclude, setTagsToExclude] = useState<string[]>([]);
+  const [tagsSet, setTagsSet] = useState<boolean>(false);
+
+  const registry = useGet<HubItemsResponse<Registry>>(
+    hubAPI`/_ui/v1/execution-environments/registries?page_size=${page_size.toString()}`
```

Comment:
this is not filtering by name

the registry selector should be a typeahead select, so that it can work with >50 repos

Reply:
It is not ment to filter by name. useRegistrySelector handles modal with PageTable that allows the filtering. I had typeahead component, but it was banned by James and others that it could affect performance of application. Instead they decided that we will add Browse button into the dropdown that will open modal with filter. I have written function that constructs this modal dynamicaly by passing columns, filters and parameters for useGet for fetching data. But framework also needs actual data for the dropdown. Im planning to return this "query" from my function, because it can be generated by the data already passed for modal. But for now, it is not implemented and so it has to be two separated calls - one for modal and other for dropdown list.